### ISSUE TITLE
Yaml plugin should not need to be run inside its own side car container

### ISSUE THREAD

**Issue by sunix** [eclipse-che/che#13555]
### Description
At the moment the yaml plugin is using `eclipse/che-theia-endpoint-runtime:next` but don't really need additional dependencies from the side car container. So it make sense to move that into the Theia-ide container.
https://github.com/eclipse/che-plugin-registry/blob/master/v3/plugins/redhat/vscode-yaml/0.4.0/meta.yaml#L15

**Comment by tsmaeder** [comment]
If you have it in the theia image, you can't turn it off. The real solution here would be to implement the idea @benoitf and @tsmaeder had a long time ago to run all plugins that use the same base image in the same container.

**Comment by benoitf** [comment]
here we could just remove the image in meta.yaml so it goes in theia container (the default)

**Comment by amisevsk** [comment]
There is an issue with not running plugins in sidecars though, if they require a significant amount of memory. Since the theia container has a relatively tight default memoryLimit, adding any plugins that require memory would also require manually increasing the memory allocated to Theia.

We currently don't have a good flow for accomodating additional plugins in the Theia container (we should e.g. take the memory limit from the plugin and add it to the theia container's default limit).

**Comment by gorkem** [comment]
@amisevsk although this is correct YAML LS works on a single file so it is relatively nimble.  Alternately you can consider running it inside the browser just like it is done for OS console.

**Comment by amisevsk** [comment]
@gorkem I agree here, but created issue https://github.com/eclipse/che/issues/14511 for discussion in any case (I'm guessing the alternative there is what we'll end up doing anyways).

**Comment by sunix** [comment]
> If you have it in the theia image, you can't turn it off. The real solution here would be to implement the idea @benoitf and @tsmaeder had a long time ago to run all plugins that use the same base image in the same container.

turn off -> just remove the plugin from the devfile, no ?

**Comment by ericwill** [comment]
I guess this is still relevant, since vscode-yaml is using the node sidecar these days.
